### PR TITLE
Moves the linalg_ext dialect to iree_linalg_ext under the iree-dialects project.

### PR DESCRIPTION
**Pull request by stellaraccident**
Mostly NFC:
* Pre-factors iree-dialects into a consistent state wrt namespaces and directory layouts.
* Moves linalg_ext to a new dialect under iree-dialects. Does some adaptation to upstream style along the way.
* Redirects everything that was using it to use the new one.
* Enables tests for iree-dialects in the cmake CI (they were not enabled) and fixes some type prefixing that had drifted. (Will follow up with enabling them in the internal builds to guard against further regression).

Non-NFC:
* When tiling, the old pass was directly generating `flow.dispatch.workgroup...` ops to query the current workgroup id. We had been planning to add those to the input dialect, so I pulled part of that patch forward.
* Since these now newly lack lowerings, I was expecting to hit some test failures that would guide me to where to adapt, but this appears to be dead code (outside of integration tests?). We'll see what the CI says.
* Can keep coding on this patch to adapt whatever is needed.

### CODE REVIEW

**Comment by stellaraccident** [comment]
Actually, I realized that the ops I changed were just part of a test pass, so I think we are good.

File: llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/TiledOpInterface.h [hunk 1]
```diff
@@ -15,17 +15,19 @@
 #include "mlir/Support/LLVM.h"
 
 /// Include the ODS generated interface header files.
-#include "iree/compiler/Dialect/LinalgExt/IR/TiledOpInterface.h.inc"
+#include "iree-dialects/Dialect/LinalgExt/IR/TiledOpInterface.h.inc"
 
 namespace mlir {
 namespace iree_compiler {
-namespace linalg_ext {
+namespace IREE {
```
Comment (benvanik) [inline llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/TiledOpInterface.h:22]: like the namespace changes; being able to use IREE:: as the universal autocomplete root is nice!

Comment (stellaraccident) [reply llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/TiledOpInterface.h:22]: Yeah, that took a lot of time to clean up and make consistent but helps a lot, imo.

**Review by benvanik — APPROVED** [review]
Looks good - I loved the way `iree.` looked for our inputs but having `iree_input.` makes it a lot clearer what it is and once we get into the main IREE dialects we nuke it anyway.

**Pull request merged**